src/Chainweb/Miner/Pact.hs

```diff
@@ -148,26 +151,42 @@ fromMinerData :: MonadThrow m => MinerData -> m Miner
 fromMinerData = decodeStrictOrThrow' . _minerData
 {-# INLINABLE fromMinerData #-}
 
+data MinerRewards = MinerRewards
+    { _minerRewards :: !(HashMap BlockHeight ParsedDecimal)
+      -- ^ The map of blockheight thresholds to miner rewards
+    , _minerRewardHeights :: !(Vector BlockHeight)
+      -- ^ A (sorted) vector of blockheights (head is most recent)
+    } deriving (Eq, Ord, Show, Generic)
```

Comment:
The state of the `Vector` isn't lost if the Node shuts down?

Reply:
It would be - that's a good point. I wonder if we should make the lookup less `head`-based and a comparison instead.